internal/services/gitserver/repo-cleaner.go

```diff
@@ -0,0 +1,197 @@
+package gitserver
+
+import (
+	"context"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"agola.io/agola/internal/services/config"
+	"agola.io/agola/internal/util"
+	"go.uber.org/zap"
+)
+
+type RepoCleaner struct {
+	dataDir          string
+	logger           *zap.SugaredLogger
+	scanTime         time.Duration
+	deleteBranchTime time.Duration
+}
+
+const (
+	DEFAULT_REPO_DELETE_BRANCH string = "720h"
+	DEFAULT_SCAN_TIME          string = "24h"
+)
+
+func NewRepoCleaner(c *config.Gitserver, l *zap.SugaredLogger) *RepoCleaner {
+	stDuration, err := time.ParseDuration(c.RepoCleaner.ScanTime)
+	if err != nil {
+		l.Error("repo-cleaner error:", err)
+		stDuration, _ = time.ParseDuration(DEFAULT_SCAN_TIME)
+	}
+
+	btDuration, err := time.ParseDuration(c.RepoCleaner.DeleteBranchAge)
+	if err != nil {
+		l.Error("repo-cleaner error:", err)
+		btDuration, _ = time.ParseDuration(DEFAULT_REPO_DELETE_BRANCH)
+	}
+
+	return &RepoCleaner{
+		dataDir:          c.DataDir,
+		logger:           l,
+		scanTime:         stDuration,
+		deleteBranchTime: btDuration,
+	}
+}
+
+func (s *RepoCleaner) Run(ctx context.Context) {
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				s.logger.Infof("repoCleaner exiting")
+
+				return
+			case <-time.After(s.scanTime):
+				if err := s.scanRepos(ctx); err != nil {
+					log.Errorf("err: %+v", err)
+				}
+			}
+		}
+	}()
+}
+
+func (s *RepoCleaner) scanRepos(ctx context.Context) error {
+	s.logger.Info("repoCleaner scanRepos start")
+
+	usersDir, err := ioutil.ReadDir(s.dataDir)
+	if err != nil {
+		return err
+	}
+
+	for _, u := range usersDir {
+		if u.IsDir() {
+			reposDir, _ := ioutil.ReadDir(s.dataDir + string(filepath.Separator) + u.Name())
```

Comment:
It's more simple to use `filepath.Join`. So:
`s.dataDir + string(filepath.Separator) + u.Name()`
will become
`filepath.Join(s.dataDir, u.Name())`